app/routes/__index/domains/index.tsx

```diff
@@ -1,22 +1,38 @@
 import { AddIcon } from '@chakra-ui/icons';
 import { Button, Container, Flex, Heading, Text } from '@chakra-ui/react';
+import type { LoaderArgs } from '@remix-run/node';
+import { Response } from '@remix-run/node';
 import { Link, useNavigate } from '@remix-run/react';
 import type { DomainsTableAction } from '~/components/domains-table';
 import DomainsTable from '~/components/domains-table';
-import DOMAINS_MOCK from '~/mocks/domains';
+import { getRecordsByUsername } from '~/models/record.server';
+import { getUser } from '~/session.server';
+import type { Record } from '@prisma/client';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await getUser(request);
+  if (!user) {
+    throw new Response('No user', { status: 500 });
+  }
+
+  return typedjson(await getRecordsByUsername(user.username));
```

Comment:
Doesn't Remix [already handle the typing](https://remix.run/docs/en/v1/guides/data-loading#basics) on loader data via `typeof loader`?

Reply:
I described this problem in the PR description, the problem that remix has is it doesn't deserialize dates, so it just returns them like strings in run-time

Reply:
@humphd what do you think about this? Do you think my justification for using this lib is fair compared to regular remix stuff?